### DISCUSSION TITLE
UI - Audiobook Player

### DISCUSSION THREAD

**Discussion by p1xelll** [advplyr/audiobookshelf-app#26] [Ideas]
I would appreciate a smaller bottom player, with fewer features. When I click on the bottom player a larger player appears with all the options.
Example:
![Android - 2](https://user-images.githubusercontent.com/81173399/139719878-acf9c672-a237-4f01-b764-967c08a97c71.png)
![Android - 14](https://user-images.githubusercontent.com/81173399/139719888-8b624b82-3b1b-4b90-bfec-8ffca033a190.png)

https://user-images.githubusercontent.com/81173399/139722113-c58e0279-a5c8-4ace-b95c-10bca001cd16.mp4

**Comment by advplyr** [comment]
I've been thinking more about the fullscreen view with all the new features. Your demo looks really good, I agree we will go with what you have there.
Few things:
1. Do you have that Zz sleep timer button as SVG?
2. What is the the previous and next track buttons doing? Right now the prev/next buttons from bluetooth will trigger the jump forward/backward 10s.
3. In your example of the sleep timer modal you have a +/- 90 min option, how does this function?
4. On the cover art aspect ratio: when I started the project I decided on using a regular book aspect ratio 1.6:1 for cover art. But the square cover art is used in audible and now adopted a lot of other places for audiobooks. Squeezing a 1.6:1 book cover in a square seemed worse then putting a square in a 1.6:1 ratio rectangle, if that makes sense. You used square in your example and it may need to be 1.6:1.

**Reply by advplyr** [reply]
Oh and did you do that animation in CSS? That animation is going to be tricky to do from scratch

**Reply by p1xelll** [reply]
I made the prototype quickly. Most are just cropped screenshots.
1. I only used the ZZ icon because I didn't have a moon icon. It's just two Z's on top of each other.
2. Rewind to the next chapter or the previous chapter. It's not necessary.
3. +/- means setting a custom length to the timer.
4. 1.6:1 ratio won't be a problem. There will just be a bigger gap.

The animations are done in Figma.

**Reply by advplyr** [reply]
Looks great. I hope you offer more designs elsewhere. The animation will be a fun challenge.

**Reply by p1xelll** [reply]
Tomorrow I'll upload a Figma project with a modified image and some changes. Animations in early access are not important. I have some more ideas not only for the app but also for the web.

**Comment by advplyr** [comment]
Just released `v0.9.16-beta` that will be available in the play store in an hour or so.

There are still some minor tweaks to be made like font sizes, spacing, and icon sizes. 

I also implemented a quick and dirty swipe down to close the player, but I think I would like this to drag with your finger like google podcast app.

Bookmarks icon is there but it is grayed out because the modal isn't built yet.

Let me know what you think, every tiny detail matters to me.

**Reply by p1xelll** [reply]
I didn't expect it to be done so quickly. Good job! I took a quick look at it and would change this:  
1. I would move the X icon lower and make it as thick as the down arrow.
![Screenshot 2021-11-02 110434](https://user-images.githubusercontent.com/81173399/139826727-b56bf2f0-c7c0-464a-ba27-8b4a35b19e6f.png)
2. I would center the speed settings modal. Right now the text is a little to the left.
![Screenshot 2021-11-02 110504](https://user-images.githubusercontent.com/81173399/139826810-778dd50f-890e-4896-8692-f6827c9f7cde.png)

**Reply by advplyr** [reply]
Thanks, I updated these and found a ton of other issues that I resolved in `v0.9.17-beta`.
Also added functionality to the bookmarks.
Should be available on the play store soon.

**Reply by p1xelll** [reply]
Now not centered modal with chapters.
![Snímka obrazovky 2021-11-03 095132](https://user-images.githubusercontent.com/81173399/140031692-8f73322a-0ef0-428d-83c6-02781268d96d.png)

**Reply by advplyr** [reply]
I can't remember if I saw this and fixed it or not, all the chapters on my audiobooks fill the entire length so I can't tell if it is centered or not.

**Comment by p1xelll** [comment]
Here's a link to [Figma](https://www.figma.com/file/Q0u2DFyFdC8bvM0CQRpugw/Untitled) though you probably don't need it.
